### PR TITLE
bedtools: update to v2.29.2

### PR DESCRIPTION
**Pull request by arjanvandervelde**
This updates bedtools from v2.28.0 to v2.29.2 and also changes its
dependence on py27-sphinx to py37-sphinx in order to prevent pulling
in python2.

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.x
Xcode 8.x

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?

- [x] checked your Portfile with `port lint`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: science/bedtools/Portfile [hunk 1]
```diff
@@ -19,11 +19,11 @@ long_description    The BEDTools utilities allow one to address common genomics
...
-                    sha256  511d8623038836ac8d1f9e260016867494b9040de65d0f0f1eb7feac7b698bbd \
-                    size    20565042
+checksums           rmd160  25eefac3397b98c1c055de92c9cbb360ee4efb59 \
+                    sha256  66d4d74000549fd5a1f939dc2bc0417d743dec72575a709213058754ff6dd146 \
+                    size    20638903
 
-depends_build       port:py27-sphinx
+depends_build       port:py37-sphinx
 
 depends_lib-append  port:zlib port:xz
```
Comment (reneeotten) [inline science/bedtools/Portfile]: not sure if this is new in this version, but when building the port on my system it also links with ```/opt/local/lib/libbz2.1.0.dylib```, but the port doesn't declare a dependency on ```bzip2```. To make sure you get "reproducible builds" one should either depend on this port (easiest way) or make sure that this dependency does not get picked up, even if it is already installed on a system.

Comment (arjanvandervelde) [reply science/bedtools/Portfile]: interesting. thanks! yeah I haven't seen this before; might be a recent change. will add.

File: science/bedtools/Portfile [hunk 2]
```diff
@@ -19,11 +19,11 @@ long_description    The BEDTools utilities allow one to address common genomics
...
 
-checksums           rmd160  0807fc84acd6d80f646ee387943ff6345203396a \
-                    sha256  511d8623038836ac8d1f9e260016867494b9040de65d0f0f1eb7feac7b698bbd \
-                    size    20565042
+checksums           rmd160  25eefac3397b98c1c055de92c9cbb360ee4efb59 \
+                    sha256  66d4d74000549fd5a1f939dc2bc0417d743dec72575a709213058754ff6dd146 \
+                    size    20638903
 
-depends_build       port:py27-sphinx
+depends_build       port:py37-sphinx
```
[the remark this answers is not in the document]
Comment (arjanvandervelde) [reply science/bedtools/Portfile]: not sure where we are on "preferred" python version these days; trying to be conservative. it doesn't really matter functionally. what is best practice here, newest?

[the remark this answers is not in the document]
Comment (reneeotten) [reply science/bedtools/Portfile]: we have already set 3.8 as the default version in the Python PG; so please go with that one.

**Comment by arjanvandervelde** [comment]
@reneeotten thanks!

File: science/bedtools/Portfile (hunk 1 above)
Comment (reneeotten) [reply science/bedtools/Portfile]: it did so already with the previous version as well, you can check with:
```otool -L /opt/local/bin/bedtools```

**Comment by reneeotten** [comment]
@arjanvandervelde thanks, looks good to me! I'll wait to see if Travis will finish without trouble, and, if so, merge this PR

**Pull request merged**